functional/ek-cert-use-ek_handle-custom-ca_certs/test.sh

```diff
@@ -0,0 +1,99 @@
+            rlRun "limeInstallIMAConfig"
+            rlRun "limeStartIMAEmulator"
+        else
+            rlDie "Test work only when TPM is emulated."
+        fi
+        #change tpm_ownerpassword    
+        rlRun "limeUpdateConf cloud_agent tpm_ownerpassword $PASSWORD_TPM"
+        #show persistent ek
+        rlRun "tpm2_getcap handles-persistent"
+        # changes authorization values for TPM objects
+        rlRun "tpm2_changeauth -c o $PASSWORD_TPM"
+        rlRun "tpm2_changeauth -c e $PASSWORD_TPM"
+        #flush previous endorsement key
+        rlRun "tpm2_evictcontrol -C o -c 0x81010001 -P $PASSWORD_TPM"
+        #manually create endorsement key from cryptographic primitives on TPM
+        rlRun "tpm2_createek -P $PASSWORD_TPM -w $PASSWORD_TPM -c 0x81010001 -G rsa"
+        # set address to which ek use
+        rlRun "limeUpdateConf cloud_agent ek_handle 0x81010001"
+        # tenant, set to true to verify ek on TPM
+        rlRun "limeUpdateConf tenant require_ek_cert true" 
+        # start keylime_verifier
+        rlRun "limeStartVerifier"
+        rlRun "limeWaitForVerifier"
+        rlRun "limeStartRegistrar"
+        rlRun "limeWaitForRegistrar"
+        rlRun "limeStartAgent"
+        rlRun "limeWaitForAgentRegistration ${AGENT_ID}"
+        # create allowlist and excludelist
+        limeCreateTestLists
+        # creating dir, where will be store CA cert for veryfi genuine of TPM 
+        rlRun "mkdir -p $CERTDIR"
+    rlPhaseEnd
+
+    rlPhaseStartTest "Fail keylime agent and verify fail of checking EK cert"
+        # create empty CA, which fail the checking of EK cert
+        rlRun "touch $CERTDIR/swtpm-localca-rootca-cert.pem $CERTDIR/issuercert.pem $CERTDIR/bundle.pem"
+        # expected to fail
+        rlRun -s "keylime_tenant -v 127.0.0.1 -t 127.0.0.1 -u $AGENT_ID --allowlist allowlist.txt --exclude excludelist.txt -f excludelist.txt -c update" 1 
+    rlPhaseEnd
+
+    rlPhaseStartTest "Add keylime agent and verify genuine of TPM via EK cert"
+        rlRun "rm -rf $CERTDIR/*"
+        # how to obtain local CA cert and how setup https://github-wiki-see.page/m/stefanberger/swtpm/wiki/Certificates-created-by-swtpm_setup
+        # cp CA cert to dir, where keylime verify genuine of TPM
+        rlRun "cp $CACERTDIR/swtpm-localca-rootca-cert.pem $CERTDIR"
+        rlRun "cp $CACERTDIR/issuercert.pem $CERTDIR"
+        rlRun "cat $CACERTDIR/issuercert.pem $CACERTDIR/swtpm-localca-rootca-cert.pem > bundle.pem"
+        rlRun "cp bundle.pem $CERTDIR"
+        rlRun -s "keylime_tenant -v 127.0.0.1 -t 127.0.0.1 -u $AGENT_ID --allowlist allowlist.txt --exclude excludelist.txt -f excludelist.txt -c update"
+        rlRun "limeWaitForAgentStatus $AGENT_ID 'Get Quote'" 
+        rlRun -s "keylime_tenant -c cvlist"
+        rlAssertGrep "{'code': 200, 'status': 'Success', 'results': {'uuids':.*'$AGENT_ID'" $rlRun_LOG -E
+    rlPhaseEnd
+
+    rlPhaseStartCleanup "Do the keylime cleanup"
+        rlRun "limeStopAgent"
```

Comment:
I am thinking whether we need to add some extra clean up steps. Maybe not.

Reply:
Maybe remove authorization values for TPM objects and evict the generated endorsement keys?